pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -4759,10 +4759,112 @@ func TestReconcileWithPipelineResults(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			runTestReconcileWithPipelineResults(t, tc.embeddedVal)
+			runTestReconcileWithFinallyResults(t, tc.embeddedVal)
 		})
 	}
 }
 
+func runTestReconcileWithFinallyResults(t *testing.T, embeddedStatus string) {
+	names.TestingSeed()
+	ps := []*v1beta1.Pipeline{parse.MustParsePipeline(t, `
+metadata:
+  name: test-pipeline
+  namespace: foo
+spec:
+  results:
+  - description: pipeline result
+    name: result
+    value: $(finally.a-task.results.aResult)
+  tasks:
+  - name: b-task
+    params:
+    - name: bParam
+      value: $(tasks.a-task.results.aResult)
```

Comment:
it looks like the task's param expression here is not valid (there's no non-finally task `a-task`). This task shouldn't need a param anyway.

Reply:
1. look if reconciler has validation for taskname not being present
2. figure out what fields of the `task` spec if required vs optional, maybe specifying task and image

Reply:
done